### PR TITLE
Added the option to use an existing terraform installation

### PR DESCRIPTION
**Pull request by lucascioffi-mb**
Even if Terraform is already installed, cf-terraforming always performs a new installation.
For single executions, that's fine.
However, when executing it multiple times or in a script, the installations can take up a lot of time, processing power and disk space.

We've created a new flag ("terraform-path") that, when supplied, skips the installation phase.
Example:
`./cf-terraforming generate --resource-type=cloudflare_zone --account <accountID> -t <token> --terraform-path '~/bin/terraform' -v`

If the flag is not supplied, the program assumes terraform is not installed and reverts to the old behaviour.

### CODE REVIEW

File: internal/app/cf-terraforming/cmd/generate.go [hunk 1]
```diff
@@ -43,36 +43,36 @@ func generateResources() func(cmd *cobra.Command, args []string) {
...
 		accountID = viper.GetString("account")
 
-		tmpDir, err := ioutil.TempDir("", "tfinstall")
-		if err != nil {
-			log.Fatal(err)
-		}
-		defer os.RemoveAll(tmpDir)
+		var execPath, workingDir string
+		workingDir = viper.GetString("terraform-install-path")
+		execPath = viper.GetString("terraform-path")
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:48]: this flag name is a little too vague as to what it actually does. how about `terraform-binary-path` instead?

```suggestion
		execPath = viper.GetString("terraform-binary-path")
```

File: internal/app/cf-terraforming/cmd/generate.go [hunk 2]
```diff
@@ -43,36 +43,36 @@ func generateResources() func(cmd *cobra.Command, args []string) {
...
+			tmpDir, err := ioutil.TempDir("", "tfinstall")
+			if err != nil {
+				log.Fatal(err)
+			}
+			defer os.RemoveAll(tmpDir)
 
-		execPath, err := installer.Install(context.Background())
-		if err != nil {
-			log.Fatalf("error installing Terraform: %s", err)
+			installer := &releases.ExactVersion{
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:58]: i've updated this section in #489 to allow a version constraint as opposed to a hard version. you'll need to update your branch to get that into here too.

File: internal/app/cf-terraforming/cmd/root.go [hunk 3]
```diff
@@ -103,11 +103,18 @@ func init() {
 
 	rootCmd.PersistentFlags().StringVar(&resourceType, "resource-type", "", "Which resource you wish to generate")
 
-	rootCmd.PersistentFlags().StringVar(&terraformInstallPath, "terraform-install-path", ".", "Path to the Terraform installation")
+	rootCmd.PersistentFlags().StringVar(&terraformInstallPath, "terraform-install-path", ".", "Path where Terraform will be installed (if you don't have it)")
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/root.go:106]: this isn't quite what this flag means. the `terraform-install-path` is for referencing a working terraform installation that isn't in the immediate directory.

Comment (jacobbednarz) [reply internal/app/cf-terraforming/cmd/root.go:106]: it uses the existing configuration (provider config, etc) as opposed to requiring you make a new one.

**Comment by jacobbednarz** [comment]
i like the concept of this however, we do need to tidy up the wording a little to differentiate the terraform installation vs the binary in use.

we'll also need to update the documentation to include the new flag in the usage section.

File: internal/app/cf-terraforming/cmd/generate.go (hunk 1 above)
Comment (lucascioffi-mb) [reply internal/app/cf-terraforming/cmd/generate.go]: Yeah, that's much better!

File: internal/app/cf-terraforming/cmd/generate.go (hunk 2 above)
Comment (lucascioffi-mb) [reply internal/app/cf-terraforming/cmd/generate.go:58]: Ok, I'll do that >:)

File: internal/app/cf-terraforming/cmd/root.go (hunk 3 above)
Comment (lucascioffi-mb) [reply internal/app/cf-terraforming/cmd/root.go:106]: Hmmm, so maybe it should say something like: `Path to an existing, initialized terraform folder. Default is "."`?

**Comment by lucascioffi-mb** [comment]
@jacobbednarz thanks for your reviews!! I'll update the readme and get back to you :)

**Comment by lucascioffi-mb** [comment]
Done! What do you think?

**Comment by jacobbednarz** [comment]
looks great, thanks for the improvement!

**Pull request merged**